theme/templates/base.html

```diff
@@ -90,6 +90,18 @@
           title="{{ SITENAME|striptags }} Tags RSS Feed"
         />
       {% endif %}
+      <script>
+        // Apply theme based on stored preference or system setting
+        const storedTheme = localStorage.getItem("theme");
+        if (storedTheme) {
+          document.documentElement.setAttribute("data-theme", storedTheme);
+        } else if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+          document.documentElement.classList.add("dark");
+          document.documentElement.setAttribute("data-theme", "dark");
+        } else {
+          document.documentElement.setAttribute("data-theme", "light");
+        }
```

Comment:
why here and not in the dedicate script files so it can be cached and saves bytes?